File: src/main/java/no/nav/domain/pensjon/regler/beregning2011/FremtidigBarnetilleggperiode.java
package no.nav.domain.pensjon.regler.beregning2011;

public class FremtidigBarnetilleggperiode extends AbstraktBarnetilleggperiode {

    /**
     * Hva perioden skal avkortes med. Årlig beløp.
     */
    private double fradragPerAr;
    /**
     * Årlig netto uten hensyn til justeringsbeløp
     */
    private int nettoPerArForJustering;
    /**
     * Årlig netto hensyntatt justeringsbeløp
     */
    private double nettoPerAr;
    /**
     * Månedlig nettobeløp
     */
    private int nettoPerMnd;
    /**
     * Hva gjenstår å utbetale i perioden
     */
    private int restTilUtbetaling;
    /**
     * Hva ville ha gjenstått å utbetale i perioden gitt ingen justering
     */
    private double restTilUtbetalingForJustering;
    /**
     * Nødvendig justering av avkortingsbeløp for perioden
     */
    private double justeringsbelopUbegrensetPerAr;
    /**
     * Hva må avkortingsbeløpet justeres med for å oppnå riktig fradrag for perioden
     */
    private double justeringsbelopPerAr;
    /**
     * Månedlig justeringsbeløp
     */
    private int justeringsbelopPerMnd;
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/FremtidigBarnetilleggperiode.kt
package no.nav.pensjon.regler.domain.beregning2011

class FremtidigBarnetilleggperiode : AbstraktBarnetilleggperiode() {
    /**
     * Hva perioden skal avkortes med. årlig beløp.
     */
    var fradragPerAr = 0.0

    /**
     * årlig netto uten hensyn til justeringsbeløp
     */
    var nettoPerArForJustering = 0

    /**
     * årlig netto hensyntatt justeringsbeløp
     */
    var nettoPerAr = 0.0

    /**
     * månedlig nettobeløp
     */
    var nettoPerMnd = 0

    /**
     * Hva gjenstår å utbetale i perioden
     */
    var restTilUtbetaling = 0

    /**
     * Hva ville ha gjenstøtt å utbetale i perioden gitt ingen justering
     */
    var restTilUtbetalingForJustering = 0.0

    /**
     * nødvendig justering av avkortingsbeløp for perioden
     */
    var justeringsbelopUbegrensetPerAr = 0.0

    /**
     * Hva må avkortingsbeløpet justeres med for å oppnå riktig fradrag for perioden
     */
    var justeringsbelopPerAr = 0.0

    /**
     * månedlig justeringsbeløp
     */
    var justeringsbelopPerMnd = 0

}